src/main.rs

```diff
@@ -97,6 +107,9 @@ async fn main() -> anyhow::Result<()> {
             )
             // Serve a static tree of files of the web UI. Must be the last item.
             .service(Files::new("/", &config.www).index_file("index.html"))
+            .default_service(
+                web::route().to(react_index)
```

Comment:
Are you trying to create a route that redirects all non existing routes to index.html?

i think you should be able to use a closure so that you can capture the config instead of loading it constantly:

``` suggestion
                web::to(|| async { NamedFile::open_async(config.www.clone().join("index.html").await.unwrap() })
```

Reply:
Yes, that's the goal! Your approach looks much cleaner, I'll give it a try after work!

Thank you for the suggestion!!

Reply:
Ok, I gave it a shot with this:
```
            .default_service(
                web::to(|| async { 
                    NamedFile::open_async(config.www.clone().join("index.html")).await
                })
            )
```

Sadly I got this error:
```
40.62 error: lifetime may not live long enough
40.62    --> src/main.rs:102:17
40.62     |
40.62 87  |       let run_server = HttpServer::new(move || {
40.62     |                                        ------- lifetime `'1` represents this closure's body
40.62 ...
40.62 102 | /                 web::to(|| async { 
40.62 103 | |                     NamedFile::open_async(config.www.clone().join("index.html")).await
40.62 104 | |                 })
40.62     | |__________________^ argument requires that `'1` must outlive `'static`
40.62     |
40.62     = note: closure implements `Fn`, so references to captured variables can't escape the closure
40.62 
40.93 error: could not compile `bmcd` (bin "bmcd") due to 1 previous error 
```